src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js

```diff
@@ -0,0 +1,865 @@
+      await render(h(ComputeModalBase, defaultModalProps))
+
+      await userEvent.click(screen.getByText('Customize'))
+      const selectMenu = await screen.getByLabelText('Application configuration')
+      await userEvent.click(selectMenu)
+      const selectOption = await screen.findByText('Custom Environment')
+      await userEvent.click(selectOption)
+
+      const imageInput = await screen.getByLabelText('Container image')
+      expect(imageInput).toBeInTheDocument()
+      const invalidImageUri = 'b'
+      await userEvent.type(imageInput, invalidImageUri)
+
+      const nextButton = await screen.findByText('Next')
+
+      verifyDisabled(nextButton)
+    })
+  })
+
+  //custom on image select with a [valid, invalid] custom image should function
+  it('custom Environment pane should work with a valid image URI ', async () => {
+    // Arrange
+    const createFunc = jest.fn()
+
+    const runtimeFunc = jest.fn(() => ({
+      details: jest.fn(),
+      create: createFunc
+    }))
+    Ajax.mockImplementation(() => ({
+      ...defaultAjaxImpl,
+      Runtimes: {
+        runtime: runtimeFunc
+      },
+      Disks: {
+        disk: () => ({
+          details: jest.fn()
+        })
+      }
+    }))
+
+    // Act and assert
+    await act(async () => {
+      await render(h(ComputeModalBase, defaultModalProps))
+
+      await userEvent.click(screen.getByText('Customize'))
+      const selectMenu = await screen.getByLabelText('Application configuration')
+      await userEvent.click(selectMenu)
+      const selectOption = await screen.findByText('Custom Environment')
+      await userEvent.click(selectOption)
+
+      const imageInput = await screen.getByLabelText('Container image')
+      expect(imageInput).toBeInTheDocument()
+      const customImageUri = 'us'
+      await fireEvent.change(imageInput, { target: { value: customImageUri } })
+
+      const nextButton = await screen.findByText('Next')
+      verifyEnabled(nextButton)
+      await userEvent.click(nextButton)
+      const unverifiedDockerWarningHeader = await screen.findByText('Unverified Docker image')
+
+      expect(unverifiedDockerWarningHeader).toBeInTheDocument()
+      const createButton = await screen.findByText('Create')
+      await userEvent.click(createButton)
+      expect(runtimeFunc).toHaveBeenCalledWith(defaultModalProps.workspace.workspace.googleProject, expect.anything())
+      expect(createFunc).toHaveBeenCalledWith(expect.objectContaining({
+        toolDockerImage: customImageUri
+      }))
+    })
+  })
+
+  // click learn more about persistent disk
+  it('should render learn more about persistent disks', async () => {
+    // Act
+    await act(async () => {
+      await render(h(ComputeModalBase, defaultModalProps))
+      const link = screen.getByText('Learn more about Persistent Disks.')
+      await userEvent.click(link)
+    })
+
+    // Assert
+    expect(screen.getByText('About persistent disk')).toBeInTheDocument()
```

Comment:
This test starts on this page and clicks the highlighted text

![image](https://user-images.githubusercontent.com/6465084/192836768-3eea2c52-c238-450b-9cbd-f79ddfb8c455.png)

Then verifies this header is present on the next page (so the original text is no longer present in the casing specified)
![image](https://user-images.githubusercontent.com/6465084/192837073-d7bd53ee-143b-4473-9f90-cd2096704a55.png)

Reply:
Right, I'm trying to make sure that the test wouldn't trivially pass because you are looking for something that is a substring of something that was already on the page (assuming the checks are case insensitive… I don't know if that's true).

Reply:
I'll add another check to be sure.